Foundation/NSHTTPCookieStorage.swift

```diff
@@ -70,20 +101,82 @@ open class HTTPCookieStorage: NSObject {
         @discussion The cookie will override an existing cookie with the
         same name, domain and path, if any.
     */
-    open func setCookie(_ cookie: HTTPCookie) { NSUnimplemented() }
+    open func setCookie(_ cookie: HTTPCookie) {
+        workQueue.sync {
+            if cookieAcceptPolicy == .never { return }
+
+            //add or replace
+            let key = cookie.domain + cookie.path + cookie.name
+            if let _ = allCookies.index(forKey: key) {
+                allCookies.updateValue(cookie, forKey: key)
+            } else {
+                allCookies[key] = cookie
+            }
+
+            //remove stale cookies, these may include the one we just added
+            let expired = allCookies.filter { (_, value) in value.expiresDate != nil && value.expiresDate!.timeIntervalSinceNow < 0}
+            for (key,_) in expired {
+                self.allCookies.removeValue(forKey: key)
+            }
+
+            updatePersistentStore()
+        }
+    }
     
+    private func createCookie(_ properties: [String: Any]) -> HTTPCookie? {
+        var cookieProperties: [HTTPCookiePropertyKey: Any] = [:]
+        for key in properties.keys {
```

Comment:
Wouldn't the use of ``for (key, value) in properties`` be more appropriate here as value is used anyway?